osu.Game/Rulesets/Objects/Types/IHasComboIndex.cs

```diff
@@ -0,0 +1,26 @@
+﻿// Copyright (c) 2007-2018 ppy Pty Ltd <contact@ppy.sh>.
+// Licensed under the MIT Licence - https://raw.githubusercontent.com/ppy/osu/master/LICENCE
+
+namespace osu.Game.Rulesets.Objects.Types
+{
+    /// <summary>
+    /// A HitObject that is part of a combo and has extended information about its position relative to other combo objects.
+    /// </summary>
+    public interface IHasComboIndex : IHasCombo
+    {
+        /// <summary>
+        /// The offset of this hitobject in the current combo.
+        /// </summary>
+        int IndexInCurrentCombo { get; set; }
+
+        /// <summary>
+        /// The offset of this hitobject in the current combo.
+        /// </summary>
+        int ComboIndex { get; set; }

```

Comment:
Why is this property duplicated?

Reply:
If this is the index of the combo as a whole in the beatmap, this should be in `IHasCombo`

Reply:
hmm, I split these out based on usage. basically there are many places we mark objects as supporting combo, but don't require any knowledge beyond the `bool NewCombo`.

would renaming this class to `IHasComboInformation` help?

Reply:
I don't see any situation where you'd want to know only of `NewCombo` and nothing else. There's not enough data there to tell you anything.

Example?